test/executor.test.js

```diff
@@ -906,6 +906,43 @@ describe('executor', function() {
     });
   });
 
+  describe('when booting with lazy connect', function() {
+    it.only('should report error', function(done) {
+      var datasource = {
+        mongodbDS: {
+          host: 'invalid-host',
+          port: 80,
+          connector: "mongodb",
+          name: "mongoDS",
+        },
+      };
+      function doBoot() {
+        boot.execute(app, someInstructions({ dataSources: datasource }));
+      };
+      // for some reason Error is uncaught
+      // Error msg is 'getaddrinfo ENOTFOUND invalid-host invalid-host:80'
+      expect(doBoot).to.throw(Error);
+    });
+
+    it('should not report error', function(done) {
+      process.env.LB_LAZYCONNECT_DATASOURCES = true;
+      var datasource = {
+        mongodbDS: {
+          host: 'invalid-host',
+          port: 80,
+          connector: "mongodb",
+          name: "mongoDS",
+        },
```

Comment:
Since this configuration is same for both tests, please refactor it into a shared variable, e.g. `DATASOURCE_THROWING_CONNECTION_ERROR`